script.js

```diff
@@ -1,4 +1,586 @@
+// };
+
+// // Create a function for reversed game mode
+// var reversedGame = function (userInput, programSelection) {
+//   var message = `Sorry, I didn't get that. <br> <br> Please type 'scissors', 'paper' or 'stone' to start playing the game with me 😊 <br> <br> You can also type 'reversed' to try our reversed mode and 'standard' for the original game.`;
+
+//   if (userInput == `reversed`) {
+//     message = `You are now in the reversed game mode. <br> <br> Please type 'scissors', 'paper' or 'stone' to start the reversed SPS game with me!`;
+//   }
+//   if (userInput == programSelection) {
+//     message = `It's a draw! <br> <br> This is the reversed game. <br> <br> We both chose ${programSelection}! <br> You didn't copy my answer did you? 😜`;
+//   }
+//   if (
+//     (userInput == `scissors` && programSelection == `paper`) ||
+//     (userInput == `paper` && programSelection == `stone`) ||
+//     (userInput == `stone` && programSelection == `scissors`)
+//   ) {
+//     message = `You lost! <br> <br> This is the reversed game. <br> <br> You chose ${userInput} and I chose ${programSelection}. <br> Better luck next time!`;
+//   }
+//   if (
+//     (userInput == `scissors` && programSelection == `stone`) ||
+//     (userInput == `stone` && programSelection == `paper`) ||
+//     (userInput == `paper` && programSelection == `scissors`)
+//   ) {
+//     message = `You won! <br> <br> This is the reversed game. <br> <br> You chose ${userInput} and I chose ${programSelection}. <br> Good job!`;
+//   }
+
+//   return message;
+// };
+
+// // Specify what happens for the 3 game modes
+// var main = function (input) {
+//   var myOutputValue = ``;
+
+//   if (currentGameMode == `waiting for player name`) {
+//     playerName = input;
+//     currentGameMode = `standard game`;
+//     myOutputValue = `Welcome, ${playerName}! <br> Please type 'scissors', 'paper' or 'stone' to start playing the game with me 😊 <br> <br> You can also type 'reversed' to try our reversed mode and 'standard' to revert back to the original game.`;
+//   } else if (currentGameMode != `waiting for player name`) {
+//     currentGameMode = changeGameMode(input);
+//     programChoice = generateChoice();
+//     if (currentGameMode == `standard game`) {
+//       myOutputValue = standardGame(input, programChoice);
+//     }
+//     if (currentGameMode == `reversed game`) {
+//       myOutputValue = reversedGame(input, programChoice);
+//     }
+//   }
+//   return myOutputValue;
+// };
+
+//======SPS Game with Korean Game Mode: Project Part 2 (More Comfortable)=======
+// Specify global variables
+var currentGameMode = `waiting for player name`;
+var lastWonPlayer = `unassigned`;
+
+// Create a function to change game mode based on input
+var changeGameMode = function (userInput) {
+  var gameMode = currentGameMode;
+  if (userInput == `standard`) {
+    gameMode = `standard game`;
+  }
+  if (userInput == `korean`) {
+    gameMode = `korean game`;
+  }
```

Comment:
There is something know as [SCREAM_CASE constants ](https://www.freecodecamp.org/news/when-to-capitalize-your-javascript-constants-4fabc0a4a4c4/)

Do try if you can to incorporate to project 2; it's simply another way to name variables that remain unchanged through out the program eg.

Declare these as global variables:
```
var STANDARD_MODE = "standard"
var KOREAN_MODE = "korean game"
```
Usage example:
```
  if (userInput == `korean`) {
    gameMode = KOREAN_MODE;
  }
```

Reply:
Good decomposition though!